### PR TITLE
Add guardduty filter resource

### PR DESCRIPTION
**Pull request by kzonov**
Here is an example of usage:

```
resource "aws_guardduty_filter" "test" {
  name        = "kzonov-test-1"
  action      = "ARCHIVE"
  detector_id = "90b549271278c0df5e089db1480dcbbf" # sandbox
  rank        = 2

  finding_criteria {
    criterion {
      field     = "region"
      values    = ["eu-west-1"]
      condition = "equals"
    }

    criterion {
      field     = "service.additionalInfo.threatListName"
      values    = ["another-thing"]
      condition = "greater_than"
    }
  }
}
```

### CODE REVIEW

**Comment by kzonov** [comment]
[Implement a GuardDuty Filter resource for Terraform](https://trello.com/c/hsVjX6Yz/1705-implement-a-guardduty-filter-resource-for-terraform)

**Comment by jtsaito** [comment]
@graffzon So far, this looks very good!

File: aws/resource_aws_guardduty_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsGuardDutyFilterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGuardDutyFilterConfig_full(),
+			},
+
```
Comment (jtsaito) [inline aws/resource_aws_guardduty_filter_test.go:59]: why space?

**Review by jtsaito — APPROVED** [review]
Great!

**Review by fmasuhr — APPROVED** [review]
Thanks a lot!

**Comment by kzonov** [comment]
As it's approved, I'm closing it and opening one for the official repo.

**Pull request closed without merging**